versioning/lockfiles/build_order.rst

```diff
@@ -0,0 +1,211 @@
+item in the first level, the second level can start, and it can build in parallel both ``pkgb/0.1@user/testing``
+and ``pkgc/0.1@user/testing``, because, as we can see in the graph, they are independent. It is necessary
+that both of them finish their build to be able to continue to the third level, that contains
+``pkgd/0.1@user/testing``, because this package depends on them.
+
+Every item in each level has 4 elements: ``[ref, package_id, context, id]``. At the moment the only
+necessary one is the first one. The ``ref`` value is the one that can be used for example in a ``install``
+command like:
+
+.. code-block:: bash
+
+    $ conan install <ref> --build=<ref> --lockfile=mylock.lock
+
+
+Defining builds
+---------------
+
+The definition of what needs to be built comes from the existing binaries plus the ``--build``
+argument in the ``conan lock create``.
+
+Let's build all the binaries for the exported packages first:
+
+.. code-block:: bash
+
+    # Build app1 and dependencies
+    $ conan install app1/0.1@user/testing --build=missing
+
+
+Now that there are binaries for all packages in the cache, capture a new lockfile, and compute its build order:
+
+.. code-block:: bash
+
+    # Capture a new lockfile, which will find all package binaries and fully lock them
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock
+    # And check whats need to be built
+    $ conan lock build-order app1.lock --json=build_order.json
+    # The build order is emtpy, nothing to build
+    []
+
+The result of this build order is empty. As the ``lock create`` managed to find existing binaries,
+everything is fully locked, nothing needs to be built.
+
+If we specify ``--build``, then the behavior is different:
+
+.. code-block:: bash
+
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock --build
+    # the lockfile will not lock the binaries
+    # And check whats need to be built
+    $ conan lock build-order app1.lock --json=build_order.json
+    # The build order is emtpy, nothing to build
+    [[["pkga/0.1@user/testing", "5ab8...1ac9", "host", "4"]], ...
+
+
+This feature is powerful when combined with ``package_id_modes``, because it can
+automatically define the minimum set of packages that needs to be built for any
+change in the dependency graph.
+
+Let's say that a new version ``pkgb/2.0@user/testing`` is created. But if we
+check the ``pkgd`` *conanfile.py* requirement, we can see that this falls outside
+of the valid version range. Then, it does not affect ``pkgd`` or ``app1`` and
+nothing needs to be built:
+
+.. code-block:: bash
+
+    $ conan create pkgb pkgb/2.0@user/testing
+    $ conan lock create --reference=app1/0.1@user/testing --lockfile-out=app1.lock
+    $ conan lock build-order app1.lock --json=build_order.json
+    [] # Empty, nothing to build, pkgb/2.0 does not become part of app1
+
+
+If on the contrary, a new ``pkgb/0.2@user/testing`` is created, and we capture a
```

Comment:
Can we use a different number? It is easy to read it wrong

Reply:
Better change the above to 1.1, it is more natural, and still outside of the range.